New feature: toggle debug for separate loggers

rmotitsuki opened the pull request:
messages on kytos console.

rmotitsuki commented:
Related to issue #939

rmotitsuki committed:
New feature list logger names and to enable/disable logger debug
 messages on kytos console.

rmotitsuki committed:
Unit tests

hdiogenes commented on kytos/core/controller.py:
```diff
@@ -132,6 +132,76 @@ def enable_logs(self):
+        for name in logging.root.manager.loggerDict:
+            if "kytos" in name:
+                loggers.append(logging.getLogger(name))
+        return loggers
```
I think a list compreehension could make the code a little more clear/concise here.
```
return [logging.getLogger(name)
        for name in logging.root.manager.loggerDict
        if 'kytos' in name]
```

hdiogenes commented on kytos/core/controller.py:
```diff
@@ -132,6 +132,76 @@ def enable_logs(self):
+    def loggers():
+        """List all logging Loggers.
+
+        Displays the Logger object, with name and logging level.
```
The method doesn't really "display" anything, just returns the object list.

hdiogenes commented on kytos/core/controller.py:
```diff
@@ -132,6 +132,76 @@ def enable_logs(self):
+            logger = logging.getLogger(name)
+            logger.setLevel(logging.NOTSET)
+        else:
+            LogManager.load_config_file(self.options.logging, False)
```
Suggestion:

```
        if not name:
            LogManager.load_config_file(self.options.logging, False)
            return

        # If logger name does not exist, raise an error.
        # otherwise the getLogger will create another Logger
        if name not in logging.root.manager.loggerDict:
            raise ValueError("Invalid logger name.")

        logger = logging.getLogger(name)
        logger.setLevel(logging.NOTSET)
```

hdiogenes commented on tests/test_core/test_controller.py:
```diff
@@ -91,3 +91,43 @@ def test_deprecation_warning(self):
+
+    def test_loggers(self):
+        """Test that all controller loggers are under kytos
+        hierachy logger"""
```
This should be generating linter warnings for multi-line docstrings.

rmotitsuki committed:
Fix code with comments from github
Change the method from debug to toggle_debug

hdiogenes commented on kytos/core/controller.py:
```diff
@@ -132,6 +132,83 @@ def enable_logs(self):
+
+    @classmethod
+    def _debug_off(cls, name):
+        """Turn on the debug.
```
Fix docstring

rmotitsuki committed:
Fix code to change debug on/off command to toggle

Remove the debug_on/debug_off to toggle debug.
Fix list comprehension code style

hdiogenes commented on kytos/core/controller.py:
```diff
@@ -132,6 +132,53 @@ def enable_logs(self):
+        if name and name not in logging.root.manager.loggerDict:
+            # A Logger name that is not declared in logging will raise an error
+            # otherwise logging would create a new Logger.
+            raise ValueError("Invalid logger name.")
```
```suggestion
            raise ValueError(f"Invalid logger name: {name}")
```

rmotitsuki committed:
Update kytos/core/controller.py

Co-Authored-By: Humberto Diógenes <hdiogenes@gmail.com>
